3.2.0 may break some Cartopy tests

### Bug report

I've started trying to get Cartopy working with the latest Matplotlib releases. Currently a lot of things fail, but there are some new things in 3.2.0rc1 that we may want to fix before releasing. Unfortunately, all those existing failures make it difficult to determine what's new, but I believe the following *are* introduced by 3.2.0rc1.

**Code for reproduction**

```
$ conda create -p /tmp/cpmpl matplotlib-base cartopy filelock
$ conda activate /tmp/cpmpl
$ pip install matplotlib==3.2.0rc1
$ pytest --pyargs cartopy -k 'test_limits_contour or test_contour_plot_bounds or test_global_contourf_wrap_no_transform or test_pcolormesh_goode_wrap or test_pcolormesh_mercator_wrap'
```

**Actual outcome**
For `test_limits_contour`:
```
E           AssertionError: 
E           Arrays are not almost equal to 6 decimals
E           
E           x and y +inf location mismatch:
E            x: array([[ inf,  inf],
E                  [-inf, -inf]])
E            y: array([[ 70., -45.],
E                  [170.,  45.]])
```

For `test_contour_plot_bounds`:
```
E       AssertionError: 
E       Arrays are not almost equal to 6 decimals
E       
E       Mismatch: 100%
E       Max absolute difference: 3230840.45
E       Max relative difference: 0.99999998
E        x: array([-0.05,  0.05, -0.05,  0.05])
E        y: array([-2763217.  ,  2681906.  ,  -263790.62,  3230840.5 ])
```

For `test_global_contourf_wrap_no_transform`:
The test image is cropped into a very small area.

For `test_pcolormesh_goode_wrap`, and `test_pcolormesh_mercator_wrap`:
The left edge of the image is visibly cropped.

**Expected outcome**

Tests pass.

**Matplotlib version**
  * Operating system: Fedora 30
  * Matplotlib version: 3.2.0rc1
  * Matplotlib backend (`print(matplotlib.get_backend())`): Agg
  * Python version: 3.7
  * Other libraries: Cartopy 0.17.0

--- a/lib/matplotlib/collections.py
+++ b/lib/matplotlib/collections.py
@@ -209,10 +209,12 @@ class Collection(artist.Artist, cm.ScalarMappable):
 
         if not transform.is_affine:
             paths = [transform.transform_path_non_affine(p) for p in paths]
-            transform = transform.get_affine()
+            # Don't convert transform to transform.get_affine() here because
+            # we may have transform.contains_branch(transData) but not
+            # transforms.get_affine().contains_branch(transData).  But later,
+            # be careful to only apply the affine part that remains.
         if not transOffset.is_affine:
             offsets = transOffset.transform_non_affine(offsets)
-            transOffset = transOffset.get_affine()
 
         if isinstance(offsets, np.ma.MaskedArray):
             offsets = offsets.filled(np.nan)
@@ -225,8 +227,8 @@ class Collection(artist.Artist, cm.ScalarMappable):
                 # offset.  LineCollections that have no offsets can
                 # also use this algorithm (like streamplot).
                 result = mpath.get_path_collection_extents(
-                    transform.frozen(), paths, self.get_transforms(),
-                    offsets, transOffset.frozen())
+                    transform.get_affine(), paths, self.get_transforms(),
+                    offsets, transOffset.get_affine().frozen())
                 return result.inverse_transformed(transData)
             if not self._offsetsNone:
                 # this is for collections that have their paths (shapes)
@@ -235,7 +237,7 @@ class Collection(artist.Artist, cm.ScalarMappable):
                 # those shapes, so we just set the limits based on their
                 # location.
                 # Finish the transform:
-                offsets = (transOffset +
+                offsets = (transOffset.get_affine() +
                            transData.inverted()).transform(offsets)
                 offsets = np.ma.masked_invalid(offsets)
                 if not offsets.mask.all():

